src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -181,6 +182,14 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             {
                 if (method.IsNonImplicitAndFromSource())
                 {
+                    var methodNode = root.FindNode(method.Locations[0].SourceSpan);
+                    if (methodNode.RawKind == syntaxFacts.SyntaxKinds.GlobalStatement)
```

Comment:
this is pretty ugly.  tbh i'm not even sure why this works.  Why does looking for the span of a method end up with the node for a GlocalStatement?

Reply:
The reason for this specific code is that later in the fixer we use this logic (`root.FindNode(method.Locations[0]...`, see `AddParameterService.cs` line 104) to find the parameter list to update. But that fails because we find a top-level statement.

The `.Locations[0]` seemed correct (it's the identifier/name of the local function). 
But the `FindNode` logic has tie-breaker rule and flag `getInnermostNodeForTie`.

I suspect the ultimate fix might be using `getInnermostNodeForTie=true`, but didn't have time to explore fully.